From 831a100ef06e9bceb306b73c43b9693840463ee9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:51 -0400
Subject: [tools] perf/tests: Fixup leak on error path in parse events test

Message-id: <1408953210-26343-224-git-send-email-jolsa@redhat.com>
Patchwork-id: 88921
O-Subject: [PATCH RHEL7.1 BZ1133083 223/322] perf tests: Fixup leak on error path in parse events test
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 2d4352c077b98215771cc081626c7a69289c4fac
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Fri Jan 3 17:30:04 2014 -0300

description
===========
We need to call the evlist destructor when failing to parse events.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/parse-events.c b/tools/perf/tests/parse-events.c
index e4ce8ae..41dee5e 100644
--- a/tools/perf/tests/parse-events.c
+++ b/tools/perf/tests/parse-events.c
@@ -1385,10 +1385,10 @@ static int test_event(struct evlist_test *e)
  if (ret) {
   pr_debug("failed to parse event '%s', err %d\n",
     e->name, ret);
-  return ret;
+ } else {
+  ret = e->check(evlist);
  }
-
- ret = e->check(evlist);
+ 
  perf_evlist__delete(evlist);
 
  return ret;
-- 
1.7.1